schutzbot/update_github_status.sh

```diff
@@ -1,5 +1,11 @@
 #!/bin/bash
 
+
+# Don't overwrite CI statuses on main branch for nightly pipelines
+if [[ "$CI_PIPELINE_SOURCE" == "schedule" && "$CI_COMMIT_BRANCH" == "main" ]]; then
```

Comment:
The same condition should apply to all branches, should it not? Happy to merge this as is, and fix in a follow-up, if that is the case.

Reply:
ATM the nightly pipeline runs only against `main` as a branch. OTOH removing the branch check doesn't seem to have negative effects. If we happen to run the nightly pipeline against a PR or another branch (e.g. when changing the pipeline) then we'll get the statuses from the pipeline related to the PR and they will not be overwritten with statuses from the scheduled execution, which is what we want. I'll push an update.